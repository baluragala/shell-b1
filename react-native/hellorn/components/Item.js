import React, { Component } from "react";
import {
  View,
  StyleSheet,
  Image,
  Text,
  Button,
  Alert,
  TouchableOpacity
} from "react-native";

import styles from "../commonStyles";

export default class Item extends Component {
  constructor() {
    super();
    this.state = { isFav: false };
  }

  toggleFav = () => {
    console.log("pressed");
    this.setState({ isFav: !this.state.isFav });
  };

  render() {
    const { name, address, category, details } = this.props.restaurant;
    return (
      <TouchableOpacity
        onPress={() => Alert.alert("pressed")}
        onLongPress={() => Alert.alert("long pressed")}
      >
        <View style={[styles.container]}>
          <Image
            source={{
              uri: "https://dummyimage.com/100x200/D3D3D3/fff.jpg&text=R"
            }}
            style={styles.thumbnail}
          />
          <View style={styles.detail}>
            <Text style={styles.title}>{name}</Text>
            <View style={{ flexDirection: "row" }}>
              {/* <MaterialIcons name="local-dining" size={30} color="green" /> */}
              <Text numberOfLines={2} style={{ fontSize: 16 }}>
                {address}
              </Text>
            </View>
            <Text style={styles.category}>{category}</Text>
            <Button
              title="Details"
              onPress={() => {
                this.props.navigation.navigate("Web", { url: details, name });
              }}
            />
          </View>
          {/* <FontAwesome
          name={this.state.isFav ? "heart" : "heart-o"}
          size={45}
          color="orange"
          style={{ flex: 2, marginLeft: 20, paddingTop: 20 }}
          onPress={this.toggleFav}
        /> */}
        </View>
      </TouchableOpacity>
    );
  }
}
