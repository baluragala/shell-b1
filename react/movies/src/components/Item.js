import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

export default function Item(props) {
  const { Title, Year, Poster, imdbID } = props.movie;
  return (
    <Link to={`/movies/${imdbID}`}>
      <div className="grid-item" onClick={() => props.onSelect(Title)}>
        <img src={Poster} alt="movie image" />
        <h3>{Title}</h3>
        <pre>{Year}</pre>
      </div>
    </Link>
  );
}

Item.propTypes = {
  movie: PropTypes.shape({
    Title: PropTypes.string.isRequired,
    Year: PropTypes.string.isRequired,
    Poster: PropTypes.string.isRequired
  }),
  onSelect: PropTypes.func.isRequired
};

Item.defaultProps = {
  onSelect: function() {
    alert("not implemented");
  }
};
