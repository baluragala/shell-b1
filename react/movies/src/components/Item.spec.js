import React from "react";
import Item from "./Item";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Link } from "react-router-dom";
import toJson from "enzyme-to-json";

Enzyme.configure({
  adapter: new Adapter()
});

describe("Item", () => {
  it("should render Link", () => {
    const movie = { imdbID: 1, Title: "mov1", Year: 2000, Poster: "post1" };
    const wrapper = shallow(<Item movie={movie} />);
    const h3 = wrapper.find("h3");
    const link = wrapper.find(Link);
    expect(h3.text()).toEqual(movie.Title);
    expect(link.prop("to")).toEqual("/movies/1");
  });

  it("should have same snapshot", () => {
    const movie = { imdbID: 1, Title: "mov1", Year: 2000, Poster: "post1" };
    const wrapper = shallow(<Item movie={movie} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
