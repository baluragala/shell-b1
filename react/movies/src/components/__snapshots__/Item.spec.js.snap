// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should have same snapshot 1`] = `
<Link
  to="/movies/1"
>
  <div
    className="grid-item"
    onClick={[Function]}
  >
    <img
      alt="movie image"
      src="post1"
    />
    <h3>
      MOV1
    </h3>
    <pre>
      2000
    </pre>
  </div>
</Link>
`;
